tqdm/_tqdm.py

```diff
@@ -402,8 +402,14 @@ def inner(df, func, *args, **kwargs):
                 # Precompute total iterations
                 total = getattr(df, 'ngroups', None)
                 if total is None:  # not grouped
-                    total = len(df) if isinstance(df, Series) \
-                        else df.size // len(df)
+                    if isinstance(df, Series):
+                        total = len(df)
+                    elif kwargs.get('axis', 0) == 0:
```

Comment:
And after thinking a bit more, I would say we should either make separate functions for separate `pandas` ones, or disallow `*args` completely. For example, `Series.apply(func, convert_dtype=True, args=(), **kwds)` has the first argument as `bool` which can be passed as `0`, and this can lead to very subtle bugs.